mautic/mautic
# Support translation for segment anniversary filters

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR is related to already merged to https://github.com/mautic/mautic/pull/6993
We've added support for anniversary, but now need support for translated anniversary filter.

If you're using english language:

- anniversary  
- anniversary -1 day

If you're using french language,  you are able to use these format:

- anniversary  
- anniversary -1 day
- anniversaire 
- anniversaire -1 day

This PR bring new DateRelativeParser (tests include) https://github.com/mautic/mautic/pull/7228/files#diff-bbfe9f8aca314124ce61104c7207459b
That classes also use in another PR https://github.com/mautic/mautic/pull/6710
Class parse string based on own dictionary or related date start with +/-. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com/7228)
2.  Create custom contact date field
3. Add two contacts. If today is 8. 2. 2019, then 
- first contact has custom date field set to 2019-02-08
- second contact has custom date field set to 2019-02-07
4. Create two segments with Custom date field equal
-  anniversary
-  anniversary -1 day
5. Wait for segments rebuild and see If contacts was added to segments correctly
Devs can also check logs on dev enviroment

I see for first segment

![image](https://user-images.githubusercontent.com/462477/52475178-037d9d80-2b9b-11e9-8d28-e9fa855701da.png)

For second segment
![image](https://user-images.githubusercontent.com/462477/52475163-f06acd80-2b9a-11e9-8d67-d12ff8e442b4.png)
6. Then remove segments and switch Mautic to french language
7. Create same two segments as above + create additional two segments with french language in filter:
-  anniversary
-  anniversary -1 day
- anniversaire
- anniversaire -1 day
8. Wait to segments rebuild and see If contacts was assigned to all four segments what you created


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/7228"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
